src/EnergyPlus/HeatPumpWaterToWaterSimple.cc

```diff
@@ -132,16 +132,16 @@ void GshpSpecs::simulate(EnergyPlusData &state,
                          Real64 &CurLoad,
                          [[maybe_unused]] bool const RunFlag)
 {
-    if (this->WWHPPlantTypeOfNum == DataPlant::TypeOf_HPWaterEFCooling) {
+    if (this->WWHPPlantTypeOfNum == DataPlant::PlantEquipmentType::HPWaterEFCooling) {
         if (calledFromLocation.loopNum == this->LoadLoopNum) { // chilled water loop
-            this->InitWatertoWaterHP(state, this->WWHPPlantTypeOfNum, this->Name, FirstHVACIteration, CurLoad);
+            this->InitWatertoWaterHP(state, static_cast<int>(this->WWHPPlantTypeOfNum), this->Name, FirstHVACIteration, CurLoad);
```

Comment:
This `static_cast` should not be necessary. Why does this function not take a `class enum` as an argument? Also, why do you have to pass the `PlantType` and `Name` as arguments when they are members and this is a member function?

Reply:
To keep track of work that's done, just gonna leave comments as I progress through the list because I'm planning to split up work on comments like these. Ignore my intermediate comments 😬

Reply:
`static_cast`s removed. TODO: fix function.